### PR TITLE
feat(container): Add initial APIs

### PR DESCRIPTION
**Pull request by kakkoyun**
Add bare-bone APIs for experimental container support.

Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

### CODE REVIEW

File: projects/fal/src/fal/api.py [hunk 1]
```diff
@@ -743,8 +736,55 @@ def function(
...
+
+
+@overload
+def function(
+    kind: Literal["container"],
+    *,
+    image: ContainerImage | None = None,
+    # Common options
+    host: FalServerlessHost = _DEFAULT_HOST,
+    server: Literal[True],
```
Comment (efiop) [inline projects/fal/src/fal/api.py:773]:
```suggestion
    serve: Literal[True],
```

Comment (kakkoyun) [reply projects/fal/src/fal/api.py]: Good catch.

File: projects/fal/tests/test_apps.py [hunk 2]
```diff
@@ -49,6 +50,23 @@ def addition_app(input: Input) -> Output:
...
+    serve=True,
+    max_concurrency=1,
+)
+def container_addition_app(input: Input) -> Output:
+    print("starting...")
+    for _ in range(input.wait_time):
+        print("sleeping...")
+        time.sleep(1)
+
+    return Output(result=input.lhs + input.rhs)
```
Comment (efiop) [inline projects/fal/tests/test_apps.py:67]: This is not being used.

**Comment by efiop** [comment]
@kakkoyun If you want to merge now, feel free to slap xfail markers (see `@pytest.mark.xfail` in the test file) to your container tests here.

**Comment by kakkoyun** [comment]
Good idea. I'll do it.

**Comment by kakkoyun** [comment]
@efiop Please merge on your own discretion.

**Review by efiop — APPROVED** [review]

**Pull request merged**